src/odemis/driver/focustracker.py

```diff
@@ -39,164 +43,162 @@ def __init__(self, name, role, channel, node_idx, **kwargs):
         model.HwComponent.__init__(self, name, role, **kwargs)
         # Connect to the CANbus and the CANopen network.
         self.network = canopen.Network()
-        self.network.connect(bustype='socketcan', channel=channel)
+        bustype = 'socketcan' if channel != 'fake' else 'virtual'
+        try:
+            self.network.connect(bustype=bustype, channel=channel)
+        except IOError as exp:
+            if exp.errno == 19:
+                raise HwError("Focus Tracker is not connected.")
+            else:
+                raise
         self.network.check()
-        self.node = canopen.RemoteNode(node_idx, os.path.dirname(odemis.__file__) + "/driver/FocusTracker.eds")
+        object_dictionary = pkg_resources.resource_filename("odemis.driver", "FocusTracker.eds")
+        self.node = canopen.RemoteNode(node_idx, object_dictionary)
         self.network.add_node(self.node)
+        if channel == 'fake':
+            # overwrite the node so we can get the fake sdo's
+            self.node = FakeRemoteFocusTracker(node_idx, object_dictionary)
```

Comment:
Do we need to create a "real" node  when the channel is "fake"? It's odd that you create a non-existent node on a virtual network, and then create another node. Couldn't use the same node all the time?